extension/visualizer/visualizer-extension.cpp

```diff
@@ -0,0 +1,125 @@
+#include "duckdb.hpp"
+#include "visualizer-extension.hpp"
+#include "duckdb/parser/parsed_data/create_pragma_function_info.hpp"
+#include "duckdb/catalog/catalog.hpp"
+#include "duckdb/common/fstream.hpp"
+#include "duckdb/main/query_profiler.hpp"
+#include "duckdb/main/client_context.hpp"
+
+namespace duckdb {
+
+static string ToHTML(ClientContext &context, const string &first_json_path, const string &second_json_path) {
+	string path = VIS_RESOURCE_DIR;
+	std::stringstream ss;
+	ss << "<!DOCTYPE <html>\n";
+	ss << "<style>\n";
+	ifstream css(path + "visualizer.css");
+	ss << css.rdbuf();
+	ss << "</style>\n";
+	ss << "<head>\n";
+	ss << "\t<meta charset=\"utf-8\">\n";
+	ss << "\t<title>Query Profile Graph for Query</title>\n";
+	ss << "</head>\n";
+	ss << "<body>\n";
+	ss << "<script>";
+	ifstream d3_js(path + "d3.js");
```

Comment:
External file loading is always difficult when it comes to distributing libraries. This would be simplified if the javascript was inlined into the source code. Could you write a script that inlines the css/javascript files into a header file (similar to `scripts/generate_csv_header.py` which generates `extension/tpch/dbgen/include/tpch_constants.hpp`)? That way no file needs to be read at run-time and the extension contains everything it needs to run independently.